client-app/src/Components/NewItemForm.tsx

```diff
@@ -15,35 +17,87 @@ interface FormErrors {
     [key: string]: string;
 }
 
+interface Option {
+    value: string;
+    label: string;
+    id?: number; // add id to Option donor/program
+}
+
 const NewItemForm: React.FC = () => {
     const [formData, setFormData] = useState<FormData>({
         itemType: '',
         currentStatus: 'Received',
         donorEmail: '',
+        donorId: null, // Initialize with null
         program: '',
+        programId: null, // Initialize with null
         imageUpload: [],
         dateDonated: '',
     });
 
-    const itemTypeOptions = [
-        { value: 'bicycle', label: 'Bicycle' },
-        { value: 'computer', label: 'Computer' },
-        // More item type options can be added here
-    ];
+    // const itemTypeOptions = [
+    //     { value: 'bicycle', label: 'Bicycle' },
+    //     { value: 'computer', label: 'Computer' },
+    //     // More item type options can be added here
+    // ];
+
+    // const donorEmailOptions = [
+    //     { value: 'email1', label: 'cooldude@gmail.com' },
+    //     { value: 'email2', label: 'cplusplushater@icloud.com' },
+    //     { value: 'email3', label: 'ISEBestBuilding@yahoo.com' },
+    // ];
+
+    // const programOptions = [
+    //     { value: 'youthProgram', label: 'Youth Program' },
+    //     { value: 'retailSales', label: 'Retail Sales' },
+    //     { value: 'recycle', label: 'Recycle' },
+    //     { value: 'earnABicycle', label: 'Earn-a-bicycle' },
+    //     { value: 'earnAComputer', label: 'Earn-a-computer' },
+    // ];
 
-    const donorEmailOptions = [
-        { value: 'email1', label: 'cooldude@gmail.com' },
-        { value: 'email2', label: 'cplusplushater@icloud.com' },
-        { value: 'email3', label: 'ISEBestBuilding@yahoo.com' },
-    ];
+    const [donorEmailOptions, setDonorEmailOptions] = useState<Option[]>([]);
+    const [programOptions, setProgramOptions] = useState<Option[]>([]);
+
+    useEffect(() => {
```

Comment:
I think it's good to split the fetching logic into separate functions or we can also use custom hooks for reusability